pkg/client/client.go

```diff
@@ -1674,6 +1674,17 @@ func (c *Client) PodCapacity(ctx context.Context) (int, error) {
 	return int(podCapacityTotal), nil
 }
 
+func (c *Client) GetTelemeterSecretToken(ctx context.Context) (string, error) {
+	currentSecret, err := c.GetSecret(ctx, c.namespace, "telemeter-client")
```

Comment:
`func (c *Client) CreateOrUpdateSecret` uses `s.GetNamespace()` that corresponds to the namespace in `assets/telemeter-client/secret.yaml`. Currently `c.namespace` and that namespace are both `openshift-monitoring`, but that could get out of sync at some point.

Reply:
Correct however I'm not so worried about the namespace as I am with the probability of the name of the secret changing. However, I have a hard time finding a more suitable solution 🤔

Reply:
What about the following?

- In pkg/manifests/manifests.go

```go
func (f *Factory) TelemeterClientSecretTemplate() (*v1.Secret, error) {
	return f.NewSecret(f.assets.MustNewAssetReader(TelemeterClientSecret))
}

// TelemeterClientSecret generates a new Secret for Telemeter client.
func (f *Factory) TelemeterClientSecret(secretData map[string][]byte) (*v1.Secret, error) {
	secretTemplate, _ := f.TelemeterClientSecretTemplate() // FIXME error handling
	return f.TelemeterClientSecretFromTemplate(secretTemplate, secretData)
}

// TelemeterClientSecret generates a new Secret for Telemeter client.
func (f *Factory) TelemeterClientSecretFromTemplate(secretTemplate *v1.Secret, secretData map[string][]byte) (*v1.Secret, error) {
	/*
	s, err := f.NewSecret(f.assets.MustNewAssetReader(TelemeterClientSecret))
	if err != nil {
		return nil, err
	}*/

	s := secretTemplate
...
```

- In pkg/client/client.go

```go
func (c *Client) GetTelemeterSecretToken(ctx context.Context, secretTemplate *v1.Secret) (map[string][]byte, error) {
	currentSecret, err := c.GetSecret(ctx, secretTemplate.Namespace, "telemeter-client")
...
```

- In pkg/tasks/telemeter.go

```go
	secretTemplate, _ := t.factory.TelemeterClientSecretTemplate() // FIXME error handling
	secretData, err := t.client.GetTelemeterSecretToken(ctx, secretTemplate)
	if err != nil {
		return errors.Wrap(err, "geting Telemeter Client Secret")
	}
...
```